src/diagnostics/Diagnostic.H

```diff
@@ -80,6 +86,8 @@ private:
     amrex::Vector<std::string> m_output_beam_names; /**< Component names to Write to output file */
     int m_nfields; /**< Number of physical fields to write */
     amrex::Vector<amrex::Geometry> m_geom_io; /**< Diagnostics geometry */
+    bool m_include_ghost_cells = false; /**< if ghost cells are included in output */
+    std::vector<bool> m_has_field; /**< if there is field output to write */
```

Comment:
please use `amrex::Vector` or include `<vector>` :)

Reply:
Sadly amrex::Vector<bool> doesn’t work because std::vector<bool> is a specialization. Should I use amrex::Vector<uint8_t> instead?

Reply:
That's confusing, is this because you need to communicate this vector somewhere?

`amrex::Vector` on CPU just derives from `std::vector`:
https://github.com/AMReX-Codes/amrex/blob/86a4e467b24343398dd8104da412531594172639/Src/Base/AMReX_Vector.H#L24-L28

Reply:
The entire type amrex::Vector<bool> does not work because std::vector<bool>, from which it inherits, is a custom specialization of std::vector<T> using bit fields. As a result std::vector<bool>::operator[] does not return bool&, instead some bit field manipulating Object.

The easy solution is to use std::vector instead of amrex::Vector, alternatively amrex::Vector<uint8_t> should also work.

Reply:
Basically amrex::Vector\<bool\> does not compile.